### PR TITLE
Add CPUTimeMetricQueryRunner to ClientQuerySegmentWalker

### PR DESCRIPTION
**Pull request by drcrallen**

### CODE REVIEW

**Comment by nishantmonu51** [comment]
:+1:

**Comment by gianm** [comment]
Looks good assuming it works- I am never really sure with the sequences

File: server/src/main/java/io/druid/server/ClientQuerySegmentWalker.java [hunk 1]
```diff
@@ -79,19 +82,34 @@ public ClientQuerySegmentWalker(
...
+        {
+          @Nullable
+          @Override
+          public ServiceMetricEvent.Builder apply(Query<T> tQuery)
+          {
+            return toolChest.makeMetricBuilder(tQuery);
+          }
+        },
+        emitter,
+        new AtomicLong(0L),
```
Comment (xvrl) [inline server/src/main/java/io/druid/server/ClientQuerySegmentWalker.java]: is there a purpose to passing this value?

File: server/src/main/java/io/druid/server/ClientQuerySegmentWalker.java [hunk 2]
```diff
@@ -79,19 +82,34 @@ public ClientQuerySegmentWalker(
...
+        true
```
Comment (xvrl) [inline server/src/main/java/io/druid/server/ClientQuerySegmentWalker.java]: do we have a config option anywhere to enable / disable cpu time reporting?

File: server/src/main/java/io/druid/server/ClientQuerySegmentWalker.java (hunk 1 above)
Comment (drcrallen) [inline server/src/main/java/io/druid/server/ClientQuerySegmentWalker.java]: I assume you mean the atomiclong(0). Yes, it goes with the reporting boolean beneath it. If the runner is set to report, then it reports whatever the value of the atomiclong accumulator. It is entirely possible to have multiple runners use the same accumulator and only report once (but that is not the case here)

File: server/src/main/java/io/druid/server/ClientQuerySegmentWalker.java (hunk 2 above)
Comment (drcrallen) [inline server/src/main/java/io/druid/server/ClientQuerySegmentWalker.java]: I don't think we have individual selectors for any of the metrics.

**Comment by himanshug** [comment]
LGTM

**Pull request merged**